tests/vendor.rs

```diff
@@ -505,3 +505,36 @@ fn switch_merged_source() {
     assert!(dir.join("vendor/.sources").exists());
     assert!(!dir.join("vendor/log").is_dir());
 }
+
+fn assert_cargo_features_on(channel: &str, success: bool) {
+    let (dir, _lock) = dir();
+
+    file(&dir, "Cargo.toml", r#"
+        cargo-features = ["test-dummy-unstable"]
+        [package]
+        name = "foo"
+        version = "0.1.0"
+        im-a-teapot = true
+
+        [dependencies]
+        log = "=0.3.5"
+    "#);
+    file(&dir, "src/lib.rs", "");
+
+    let out = vendor(&dir)
+        .env("__CARGO_TEST_CHANNEL_OVERRIDE_DO_NOT_USE_THIS", channel)
```

Comment:
Heh the name here was supposed to discourage use...

In any case I'm fine not having a test for this, while it's good to know that it works I'd rather not be testing against all these Cargo internals